route/table.go

```diff
@@ -0,0 +1,171 @@
+package route
+
+import (
+	"bytes"
+	"container/list"
+	"encoding/binary"
+	"math/bits"
+	"net"
+	"sort"
+)
+
+const idLength = 160
+const bytesLength = idLength / 8
+const bucketSize = 20
+
+type bucket struct{ *list.List }
+type candidates []Contact
+
+// Table implements a routing table according to the Kademlia specification.
+type Table [idLength]*bucket
+
+// NodeID represents a node's ID.
+type NodeID [bytesLength]byte
+
+// Contact contains the NodeID and an UDP address.
+type Contact struct {
+	NodeID   NodeID
+	Address  net.UDPAddr
+	distance uint64
```

Comment:
I see that there is a distance in the Contact. Further down I also see some logic that I interpret as the bucket sorting with the regard to the distance saved in each contact. 
My understanding of the paper is that the buckets should be sorted according to most recent point of communication. I.E an time metric, not an binary distance metric, do you agree?
This might've been something you discussed today that I've thereby missed. I norder for the network to retain the property of increasing robustness as time passes I think we need to save the latest communication with a contact in the Contact struct. Also sort the bucket with respect to most recent communication.

Hopefully this tired comment makes sense tomorrow too. Good Night my fellows. :sleeping:

Reply:
Ah no, I see that the sort is for when doing FindNode RPC and sorting the response. 
And that there is no need for a  time metric in the contact since you just move the most recently handled contact to the top of the list. Quite Exquisite solution yes! :+1:

Reply:
Resolved by discussion